File: packages/android/src/main/java/com/kozmos/components/NavigationItem/NavigationItem.figma.kt
package com.kozmos.components.navigationitem

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material.icons.filled.Home
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=861-7297")
class KozmosNavigationItemConnect {
    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Overview"

    @FigmaProperty(FigmaType.Enum, "Placement")
    val placement: KozmosNavigationItemPlacement = Figma.mapping(
        "Top" to KozmosNavigationItemPlacement.Top,
        "Side" to KozmosNavigationItemPlacement.Side,
        "Rail" to KozmosNavigationItemPlacement.Rail
    )

    @FigmaProperty(FigmaType.Enum, "Density")
    val density: KozmosNavigationItemDensity = Figma.mapping(
        "Default" to KozmosNavigationItemDensity.Default,
        "Compact" to KozmosNavigationItemDensity.Compact
    )

    @FigmaProperty(FigmaType.Enum, "Content")
    val content: KozmosNavigationItemContent = Figma.mapping(
        "Label" to KozmosNavigationItemContent.Label,
        "Icon Label" to KozmosNavigationItemContent.IconLabel,
        "Icon Only" to KozmosNavigationItemContent.IconOnly,
        "Badge" to KozmosNavigationItemContent.Badge,
        "Trailing" to KozmosNavigationItemContent.Trailing
    )

    @FigmaProperty(FigmaType.Enum, "State")
    val state: KozmosNavigationItemState = Figma.mapping(
        "Default" to KozmosNavigationItemState.Default,
        "Hover" to KozmosNavigationItemState.Hover,
        "Selected" to KozmosNavigationItemState.Selected,
        "Focus" to KozmosNavigationItemState.Focus,
        "Disabled" to KozmosNavigationItemState.Disabled
    )

    @FigmaProperty(FigmaType.Boolean, "Focus Visible")
    val focusVisible: Boolean = false

    @Composable
    fun ComponentExample() {
        val iconSlot: (@Composable () -> Unit)? =
            if (content != KozmosNavigationItemContent.Label) {
                { Icon(Icons.Default.Home, contentDescription = null) }
            } else {
                null
            }
        val badgeSlot: (@Composable () -> Unit)? =
            if (content == KozmosNavigationItemContent.Badge) {
                { Text("3") }
            } else {
                null
            }
        val trailingSlot: (@Composable () -> Unit)? =
            if (content == KozmosNavigationItemContent.Trailing) {
                {
                    Icon(
                        Icons.AutoMirrored.Filled.KeyboardArrowRight,
                        contentDescription = null
                    )
                }
            } else {
                null
            }

        KozmosNavigationItem(
            label = if (content == KozmosNavigationItemContent.IconOnly) null else label,
            placement = placement,
            density = density,
            content = content,
            state = state,
            selected = state == KozmosNavigationItemState.Selected,
            enabled = state != KozmosNavigationItemState.Disabled,
            focusVisible = focusVisible,
            icon = iconSlot,
            badge = badgeSlot,
            trailing = trailingSlot
        )
    }
}

File: packages/ios/Sources/Components/NavigationItem/NavigationItem.figma.swift
import SwiftUI
import Figma

struct KozmosNavigationItemConnect: FigmaConnect {
    let component = KozmosNavigationItem.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=861-7297"

    @FigmaString("Label Text")
    var label: String = "Overview"

    @FigmaEnum(
        "Placement",
        mapping: [
            "Top": KozmosNavigationItemPlacement.top,
            "Side": KozmosNavigationItemPlacement.side,
            "Rail": KozmosNavigationItemPlacement.rail
        ]
    )
    var placement: KozmosNavigationItemPlacement = .side

    @FigmaEnum(
        "Density",
        mapping: [
            "Default": KozmosNavigationItemDensity.default,
            "Compact": KozmosNavigationItemDensity.compact
        ]
    )
    var density: KozmosNavigationItemDensity = .default

    @FigmaEnum(
        "Content",
        mapping: [
            "Label": KozmosNavigationItemContent.label,
            "Icon Label": KozmosNavigationItemContent.iconLabel,
            "Icon Only": KozmosNavigationItemContent.iconOnly,
            "Badge": KozmosNavigationItemContent.badge,
            "Trailing": KozmosNavigationItemContent.trailing
        ]
    )
    var content: KozmosNavigationItemContent = .label

    @FigmaEnum(
        "State",
        mapping: [
            "Default": KozmosNavigationItemState.default,
            "Hover": KozmosNavigationItemState.hover,
            "Selected": KozmosNavigationItemState.selected,
            "Focus": KozmosNavigationItemState.focus,
            "Disabled": KozmosNavigationItemState.disabled
        ]
    )
    var state: KozmosNavigationItemState = .default

    @FigmaBoolean("Focus Visible")
    var focusVisible: Bool = false

    var body: some View {
        switch self.content {
        case .label:
            KozmosNavigationItem(
                label: self.label,
                placement: self.placement,
                density: self.density,
                content: .label,
                state: self.state,
                selected: self.state == .selected,
                disabled: self.state == .disabled,
                focusVisible: self.focusVisible
            )
        case .iconOnly:
            KozmosNavigationItem(
                label: nil,
                placement: self.placement,
                density: self.density,
                content: .iconOnly,
                state: self.state,
                selected: self.state == .selected,
                disabled: self.state == .disabled,
                focusVisible: self.focusVisible
            ) {
                Image(systemName: "house")
            }
        case .badge:
            KozmosNavigationItem(
                label: self.label,
                placement: self.placement,
                density: self.density,
                state: self.state,
                selected: self.state == .selected,
                disabled: self.state == .disabled,
                focusVisible: self.focusVisible,
                icon: {
                    Image(systemName: "house")
                },
                badge: {
                    Text("3")
                }
            )
        case .trailing:
            KozmosNavigationItem(
                label: self.label,
                placement: self.placement,
                density: self.density,
                state: self.state,
                selected: self.state == .selected,
                disabled: self.state == .disabled,
                focusVisible: self.focusVisible,
                icon: {
                    Image(systemName: "gearshape")
                },
                trailing: {
                    Image(systemName: "chevron.right")
                }
            )
        case .iconLabel:
            KozmosNavigationItem(
                label: self.label,
                placement: self.placement,
                density: self.density,
                content: .iconLabel,
                state: self.state,
                selected: self.state == .selected,
                disabled: self.state == .disabled,
                focusVisible: self.focusVisible
            ) {
                Image(systemName: "house")
            }
        }
    }
}
